apollo-router/src/plugins/external.rs

```diff
@@ -57,260 +187,250 @@ struct BaseConf {
     sdl: bool,
 }
 
+/// What information is passed to a subgraph request/response stage
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
-#[serde(default)]
-struct RouterStage {
-    /// The request configuration
-    request: Option<BaseConf>,
-    /// The response configuration
-    response: Option<BaseConf>,
+#[serde(default, deny_unknown_fields)]
+struct SubgraphConf {
+    /// Send the headers
+    headers: bool,
+    /// Send the context
+    context: bool,
+    /// Send the body
+    body: bool,
+    /// Send the service name
+    service: bool,
+    /// Send the subgraph URI
+    uri: bool,
+    /// Send the service name
+    service_name: bool,
 }
 
 /// The stages request/response configuration
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, Serialize, JsonSchema)]
-#[serde(default)]
+#[serde(default, deny_unknown_fields)]
 struct Stages {
     /// The router stage
-    router: Option<RouterStage>,
+    router: RouterStage,
+    /// The subgraph stage
+    subgraph: SubgraphStage,
```

Comment:
maybe nest this under an `all` key, to make it clear that it will apply to all subgraph requests, and allow for per subgraph configuration in the future

Reply:
fixed in [ea4d9a6](https://github.com/apollographql/router/pull/2524/commits/ea4d9a6804801f8a10dc373b66b28f462c7cdf24)